nutanix/data_source_nutanix_floating_ip_test.go

```diff
@@ -0,0 +1,80 @@
+package nutanix
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccNutanixFloatingIPDataSource_basic(t *testing.T) {
+	r := randIntBetween(31, 40)
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFloatingIPDataSourceConfig(r),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.nutanix_floating_ip.test", "spec.0.resources.0.external_subnet_reference.kind", "subnet"),
+					resource.TestCheckResourceAttr("data.nutanix_floating_ip.test", "spec.0.resources.0.vpc_reference.kind", "vpc"),
+				),
```

Comment:
check more details if possible which are unique for all entities like names, etc. Also some IPs, just to confirm we are not omitting any Set.